### PR TITLE
Pin mpi4py for 4.0.x

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mwittgen** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ fftw:
 fortran_compiler:
 - gfortran
 fortran_compiler_version:
-- '10'
+- '12'
```
Comment (erykoff) [inline .ci_support/linux_64_.yaml:36]: Undo this and the next fortran change from the rerender please.

File: .ci_support/osx_64_.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ fftw:
 fortran_compiler:
 - gfortran
 fortran_compiler_version:
-- '9'
+- '12'
```
Comment (erykoff) [inline .ci_support/osx_64_.yaml:34]: And here

File: .ci_support/osx_arm64_.yaml [hunk 3]
```diff
@@ -31,7 +31,7 @@ fftw:
 fortran_compiler:
 - gfortran
 fortran_compiler_version:
-- '11'
+- '12'
```
Comment (erykoff) [inline .ci_support/osx_arm64_.yaml:34]: And here

**Pull request closed without merging**